jerry-core/ecma/base/ecma-helpers-conversion.c

```diff
@@ -294,8 +294,12 @@ static ecma_number_t
 ecma_utf8_string_to_number_by_radix (const lit_utf8_byte_t *str_p, /**< utf-8 string */
                                      const lit_utf8_byte_t *end_p, /**< end of utf-8 string  */
                                      uint32_t radix) /**< radix */
+
 {
+  bool allow_underscore = (radix % 2) ? radix-- : false;
```

Comment:
I find this strange, while it is true that it works in this case. I would prefer:

```
bool allow_underscore = (radix & 0x1);
radix &= (uint32_t) ~0x1;
```
Maybe we could add a define constant for 0x1

Furthermore guard this with JERRY_ESNEXT which makes the `JERRY_UNUSED (allow_underscore)` below unnecessary.